core/scripts/vrfv2/testnet/setup-envs/main.go

```diff
@@ -0,0 +1,448 @@
+func printETHKeyData(ethKeys []presenters.ETHKeyResource) {
+	fmt.Println("------------- NODE INFORMATION -------------")
+	for _, ethKey := range ethKeys {
+		fmt.Println("-----------ETH Key-----------")
+		fmt.Println("Address: ", ethKey.Address)
+		fmt.Println("MaxGasPriceWei: ", ethKey.MaxGasPriceWei)
+		fmt.Println("EthBalance: ", ethKey.EthBalance)
+		fmt.Println("NextNonce: ", ethKey.NextNonce)
+		fmt.Println("-----------------------------")
+	}
+}
+
+func mapEthKeysToStringArr(ethKeys []presenters.ETHKeyResource) []string {
+	var ethKeysString []string
+	for _, ethKey := range ethKeys {
+		ethKeysString = append(ethKeysString, ethKey.Address)
+	}
+	return ethKeysString
+}
+
+func mapVrfKeysToStringArr(vrfKeys []presenters.VRFKeyResource) []string {
+	var vrfKeysString []string
+	for _, vrfKey := range vrfKeys {
+		vrfKeysString = append(vrfKeysString, vrfKey.Uncompressed)
+	}
+	return vrfKeysString
+}
+
+func printVRFKeyData(vrfKeys []presenters.VRFKeyResource) {
+	fmt.Println("Number of VRF Keys on the node: ", len(vrfKeys))
+	fmt.Println("------------- NODE INFORMATION -------------")
+	for _, vrfKey := range vrfKeys {
+		fmt.Println("-----------VRF Key-----------")
+		fmt.Println("Compressed: ", vrfKey.Compressed)
+		fmt.Println("Uncompressed: ", vrfKey.Uncompressed)
+		fmt.Println("Hash: ", vrfKey.Hash)
+		fmt.Println("-----------------------------")
+	}
+}
+
+func connectToNode(nodeURL *string, output *bytes.Buffer, credFile string) (*clcmd.Shell, *cli.App) {
+	client, app := newApp(*nodeURL, output)
+	// login first to establish the session
+	fmt.Println("logging in to:", *nodeURL)
+	loginFs := flag.NewFlagSet("test", flag.ContinueOnError)
+	loginFs.String("file", credFile, "")
+	loginFs.Bool("bypass-version-check", true, "")
+	loginCtx := cli.NewContext(app, loginFs, nil)
+	err := client.RemoteLogin(loginCtx)
+	helpers.PanicErr(err)
+	output.Reset()
+	fmt.Println()
+	return client, app
+}
+
+func createVRFKeyIfNeeded(client *clcmd.Shell, app *cli.App, output *bytes.Buffer, numVRFKeys *int, nodeURL *string) []presenters.VRFKeyResource {
+	var allVRFKeys []presenters.VRFKeyResource
+	var newKeys []presenters.VRFKeyResource
+
+	vrfKeys := getVRFKeys(client, app, output)
+
+	switch {
+	case len(vrfKeys) == *numVRFKeys:
+		fmt.Println(checkMarkEmoji, "found", len(vrfKeys), "vrf keys on", nodeURL)
+	case len(vrfKeys) > *numVRFKeys:
+		fmt.Println(checkMarkEmoji, "found", len(vrfKeys), "vrf keys on", nodeURL, " which is more than expected")
+		os.Exit(1)
+	default:
+		fmt.Println(xEmoji, "found only", len(vrfKeys), "vrf keys on", nodeURL, ", creating",
+			*numVRFKeys-len(vrfKeys), "more")
+		toCreate := *numVRFKeys - len(vrfKeys)
+		for i := 0; i < toCreate; i++ {
+			output.Reset()
+			newKey := createVRFKey(client, app, output)
+			newKeys = append(newKeys, newKey)
+			fmt.Println("NEW VRF KEYS:", strings.Join(func() (r []string) {
+				for _, k := range newKeys {
+					r = append(r, k.Uncompressed)
+				}
+				return
+			}(), ", "))
```

Comment:
This is being run on every iteration of the loop, maybe only need to run it one afterwards?